### PR TITLE
Support latest opentelemetry version for tracing sdk

### PR DESCRIPTION
**Pull request by samvaity**
This PR adds support to the latest `opentelemetry-api` , `0.14.1` for tracing sdk `azure-core-tracing-opentelemetry` and Azure Monitor OpenTelemetry Exporter client library for Java.

Closes : #18512

### CODE REVIEW

File: sdk/core/azure-core-tracing-opentelemetry/src/main/java/com/azure/core/tracing/opentelemetry/OpenTelemetryTracer.java [hunk 1]
```diff
@@ -29,12 +31,9 @@
  * requests.
  */
 public class OpenTelemetryTracer implements com.azure.core.util.tracing.Tracer {
-    private static final Tracer TRACER = OpenTelemetry.getTracerProvider().get("Azure-OpenTelemetry");
-
+    private static final Tracer TRACER = GlobalOpenTelemetry.getTracer("Azure-OpenTelemetry");
     // standard attributes with AMQP request
     static final String AZ_NAMESPACE_KEY = "az.namespace";
-    static final String MESSAGE_BUS_DESTINATION = "message_bus.destination";
-    static final String PEER_ENDPOINT = "peer.address";
```
Comment (samvaity) [inline sdk/core/azure-core-tracing-opentelemetry/src/main/java/com/azure/core/tracing/opentelemetry/OpenTelemetryTracer.java:37]: These were updated to follow https://github.com/open-telemetry/opentelemetry-java/blob/master/semconv/src/main/java/io/opentelemetry/api/trace/attributes/SemanticAttributes.java
Should we do the same in the exporter?
message_bus.destination ->  messaging.destination

File: sdk/core/azure-core-tracing-opentelemetry/src/main/java/com/azure/core/tracing/opentelemetry/implementation/HttpTraceUtil.java [hunk 2]
```diff
@@ -37,89 +47,144 @@ private HttpTraceUtil() { }
      * href="https://github.com/census-instrumentation/opencensus-specs/blob/master/trace/HTTP.md">OpenTelemetry
      * Specs</a>.
      *
+     *
+     * @param span
```
Comment (samvaity) [inline sdk/core/azure-core-tracing-opentelemetry/src/main/java/com/azure/core/tracing/opentelemetry/implementation/HttpTraceUtil.java:51]: Add javadoc

File: sdk/core/azure-core-tracing-opentelemetry/CHANGELOG.md [hunk 3]
```diff
@@ -2,6 +2,9 @@
 
 ## 1.0.0-beta.7 (Unreleased)
 
+### Dependency Updates
+- Added support for `opentelemetry-api` latest `0.14.1` version.
+For more detailed information, look [here](https://github.com/open-telemetry/opentelemetry-java/blob/master/CHANGELOG.md#version-0130---2020-12-17)
```
Comment (srnagar) [inline sdk/core/azure-core-tracing-opentelemetry/CHANGELOG.md:7]:
```suggestion
More detailed information about the new OpenTelemetry API version can be found in [OpenTelemetry changelog](https://github.com/open-telemetry/opentelemetry-java/blob/master/CHANGELOG.md#version-0130---2020-12-17)
```

File: pom.xml [hunk 4]
```diff
@@ -381,7 +381,7 @@
           <plugin>
             <groupId>org.apache.maven.plugins</groupId>
             <artifactId>maven-checkstyle-plugin</artifactId>
-            <version>3.1.0</version> <!-- {x-version-update;org.apache.maven.plugins:maven-checkstyle-plugin;external_dependency} -->
+            <version>3.1.1</version> <!-- {x-version-update;org.apache.maven.plugins:maven-checkstyle-plugin;external_dependency} -->
```
Comment (srnagar) [inline pom.xml:384]: This should be updated in `external_dependencies.txt`

File: sdk/monitor/azure-opentelemetry-exporter-azuremonitor/CHANGELOG.md [hunk 5]
```diff
@@ -2,12 +2,18 @@
...
+- Added support for `opentelemetry-api` and `opentelemetry-sdk` latest `0.14.1` version.
 
 ## 1.0.0-beta.2 (2021-01-12)
 ### Breaking changes
 - Renamed artifact to `azure-opentelemetry-exporter-azuremonitor`.
 - Replaced `instrumentationKey()` with `connectionString()` in the `AzureMonitorExporterBuilder`.
 
+### Dependency updates
+- Updated `opentelemetry-api` and `opentelemetry-sdk` version to `0.13.1` 
+  and `azure-core-tracing-opentelemetry` to `1.0.0-beta.7`
```
Comment (srnagar) [inline sdk/monitor/azure-opentelemetry-exporter-azuremonitor/CHANGELOG.md:15]: I don't think this dependency was updated for 1.0.0-beta.2 release.

File: sdk/monitor/azure-opentelemetry-exporter-azuremonitor/CHANGELOG.md [hunk 6]
```diff
@@ -2,12 +2,18 @@
 
 ## 1.0.0-beta.3 (Unreleased)
 
+### Dependency Updates
+- Added support for `opentelemetry-api` and `opentelemetry-sdk` latest `0.14.1` version.
```
Comment (srnagar) [inline sdk/monitor/azure-opentelemetry-exporter-azuremonitor/CHANGELOG.md:6]:
```suggestion
- Updated versions of `opentelemetry-api` and `opentelemetry-sdk` to `0.14.1`.
```

File: sdk/monitor/azure-opentelemetry-exporter-azuremonitor/src/main/java/module-info.java [hunk 7]
```diff
@@ -2,11 +2,14 @@
...
 module com.azure.opentelemetry.exporter.azuremonitor {
+
     requires transitive com.azure.core;
+
     requires transitive io.opentelemetry.sdk;
-    requires transitive io.opentelemetry.sdk.tracing;
+    requires transitive io.opentelemetry.sdk.trace;
     requires transitive io.opentelemetry.sdk.common;
     requires transitive io.opentelemetry.api;
+    requires io.opentelemetry.semconv;
```
Comment (srnagar) [inline sdk/monitor/azure-opentelemetry-exporter-azuremonitor/src/main/java/module-info.java:12]: Should this be `transitive` as well?

**Review by alzimmermsft — APPROVED** [review]

File: pom.xml (hunk 4 above)
Comment (samvaity) [reply pom.xml:384]: This correct - https://github.com/Azure/azure-sdk-for-java/blob/master/eng/versioning/external_dependencies.txt#L232 ?

File: sdk/core/azure-core-tracing-opentelemetry/src/main/java/com/azure/core/tracing/opentelemetry/OpenTelemetryTracer.java (hunk 1 above)
Comment (srnagar) [reply sdk/core/azure-core-tracing-opentelemetry/src/main/java/com/azure/core/tracing/opentelemetry/OpenTelemetryTracer.java:37]: Yes, we should use the same name in exporter too.

**Review by srnagar — APPROVED** [review]
LGTM!

**Pull request merged**